### ISSUE TITLE
Prop changes don't trigger component redraws

### ISSUE THREAD

**Issue by paulodeon** [psychobolt/react-paperjs#20]
It seems that prop changes to Container components don't trigger a redraw while adding and removing components does trigger a redraw.

See below a minimal test case:

Clicking the button shows the third rectangle but doesn't change the size of the first one.

```javascript
import React, { useEffect, useState, useRef, forwardRef, Fragment } from "react"
import { PaperContainer, Layer, Group, Rectangle, Circle, Line, PointText, Arc, Path, renderWithPaperScope } from '@psychobolt/react-paperjs'

export default function TestCanvas() {
  const [size, setSize] = useState({ width: '100%', height: '100%' })
  const [step, setStep] = useState('a')
  const container = useRef()

  const onClick = () => {
    setStep('b')
  }

  return (
    <Fragment>
      <button onClick={onClick}>Test Button</button>

      <PaperContainer
        ref={container}
        canvasProps={{ resize: 'true', style: Object.assign({ backgroundColor: '#F9FAFB' }, size) }}
      >
        <Rectangle
          position={[100, 100]}
          size={step === 'b' ? [300, 100] : [100, 100]}
          fillColor="white"
          radius={20}
          strokeWidth={2}
          strokeColor="#C4B5FD"
        />
        <Group>
          <Rectangle
            position={[200, 200]}
            size={[100, 100]}
            fillColor="white"
            radius={20}
            strokeWidth={2}
            strokeColor="#C4B5FD"
          />
          {step === 'b' && (
            <Rectangle
              position={[300, 300]}
              size={[100, 100]}
              fillColor="white"
              radius={20}
              strokeWidth={2}
            strokeColor="#C4B5FD"
            />
          )}
        </Group>
      </PaperContainer>
  </Fragment>
  )
}
```

**Comment by psychobolt** [comment]
Will look into it when I have a chance. Could be that it's not being detected as a change through [`diffProps`](https://github.com/psychobolt/react-paperjs/blob/0021f277f28fb027831715958e91720a8d9d24d4/src/Paper.component.js#L15), or we need special handling for `size` prop for `Rectangle` in [`updateProps`](https://github.com/psychobolt/react-paperjs/blob/0021f277f28fb027831715958e91720a8d9d24d4/src/Paper.component.js#L41).

**Comment by psychobolt** [comment]
@paulodeon I published v1.0.1, let me know if this resolves your issue.

**Comment by paulodeon** [comment]
The rectangle does now resize, however the rounded corners are now stretched.

That looks like something that would happen on a resize rather than a redraw?

![image](https://user-images.githubusercontent.com/29513/115583583-ef2ca380-a2c1-11eb-9a9e-582b1457cbaa.png)

**Comment by paulodeon** [comment]
As an aside I had to pull the package locally and add it to my package.json as a local file. 

The test above ran ok but I got a few of these errors: "Invalid hook call"

Don't know if the issue above is related or not, but if we do another round of testing it would be good to know the correct way to include the unpublished package

**Comment by psychobolt** [comment]
@paulodeon Yes, a redraw (unmount/remount) would pretty much reconstruct the Paper instance and you'll get an expected result. 

However, ideally, it is computationally efficient to do resizing based on props (if it's a few changes), so we want to handle cases where reconciler handles resizing. This is something that one would trigger using one or two Paper API call.

This library is still at its primitive state, so in some cases different type of change haven't been implemented yet (and some of it is very experimental). 

On the issue, I think the correct solution is to use the [Rectangle's API](https://github.com/paperjs/paper.js/blob/3ef3ca66d5a615df53f001331081396ae701c276/src/basic/Rectangle.js#L287) rather than updating the [instance's bound](https://github.com/psychobolt/react-paperjs/blob/e6f1e5141d1b7782010fa241db9ba1c730310845/src/Paper.component.js#L47). However, I'm still not sure why the [instance constructor](https://github.com/psychobolt/react-paperjs/blob/e6f1e5141d1b7782010fa241db9ba1c730310845/src/Paper.types.js#L36) returns an instance of [Path](http://paperjs.org/reference/path/) and not [Rectangle](http://paperjs.org/reference/rectangle/). I will investigate further...

**Comment by psychobolt** [comment]
> As an aside I had to pull the package locally and add it to my package.json as a local file.
> 
> The test above ran ok but I got a few of these errors: "Invalid hook call"
> 
> Don't know if the issue above is related or not, but if we do another round of testing it would be good to know the correct way to include the unpublished package

Ah, sorry. I didn't realize the package was not published. Please install the [latest](https://www.npmjs.com/package/@psychobolt/react-paperjs/v/1.0.1)...
